menas/src/main/scala/za/co/absa/enceladus/rest/controllers/RunController.scala

```diff
@@ -18,51 +18,88 @@ package za.co.absa.enceladus.rest.controllers
+  @GetMapping(Array("/{datasetName}/{datasetVersion}/{runId}"))
+  @ResponseStatus(HttpStatus.OK)
+  def getRun(@PathVariable datasetName: String,
+             @PathVariable datasetVersion: Int,
+             @PathVariable runId: Int): CompletableFuture[Run] = {
+    runService.getRun(datasetName, datasetVersion, runId)
+  }
 
   @GetMapping(Array("/{datasetName}/{datasetVersion}/latest"))
-  def getLatestRun(): CompletableFuture[Run] = Future(null)
-
-  @GetMapping(Array("/{datasetName}/{datasetVersion}/{runId}"))
-  def getRun(): CompletableFuture[Run] = Future(null)
+  @ResponseStatus(HttpStatus.OK)
+  def getLatestRun(@PathVariable datasetName: String,
+                   @PathVariable datasetVersion: Int): CompletableFuture[Run] = {
+    runService.getLatestRun(datasetName, datasetVersion)
+  }
+
+  @GetMapping(Array("/splineUrl/{datasetName}/{datasetVersion}/{runId}"))
+  @ResponseStatus(HttpStatus.OK)
+  def getSplineUrl(@PathVariable datasetName: String,
+                   @PathVariable datasetVersion: Int,
+                   @PathVariable runId: Int): CompletableFuture[String] = {
+    runService.getSplineUrl(datasetName, datasetVersion, runId)
+  }
 
   @PostMapping(Array("/create"))
-  def create(): CompletableFuture[Run] = Future(null)
+  @ResponseStatus(HttpStatus.CREATED)
+  def create(@RequestBody run: Run,
+             @AuthenticationPrincipal principal: UserDetails): CompletableFuture[Run] = {
+    runService.create(run, principal.getUsername)
+  }
 
   @PostMapping(Array("/addCheckpoint/{uniqueId}"))
-  def addCheckpoint(): CompletableFuture[Checkpoint] = Future(null)
+  @ResponseStatus(HttpStatus.OK)
+  def addCheckpoint(@PathVariable uniqueId: String,
+                    @RequestBody checkpoint: Checkpoint): CompletableFuture[Run] = {
+    runService.addCheckpoint(uniqueId, checkpoint)
+  }
 
   @PostMapping(Array("/updateControlMeasure/{uniqueId}"))
-  def updateControlMeasure(): CompletableFuture[ControlMeasure] = Future(null)
+  @ResponseStatus(HttpStatus.OK)
+  def updateControlMeasure(@PathVariable uniqueId: String,
+                           @RequestBody controlMeasure: ControlMeasure): CompletableFuture[Run] = {
+    runService.updateControlMeasure(uniqueId, controlMeasure)
+  }
 
   @PostMapping(Array("/updateSplineReference/{uniqueId}"))
-  def updateSplineReference(): CompletableFuture[SplineReference] = Future(null)
+  @ResponseStatus(HttpStatus.OK)
+  def updateSplineReference(@PathVariable uniqueId: String,
+                            @RequestBody splineReference: SplineReference): CompletableFuture[Run] = {
+    runService.updateSplineReference(uniqueId, splineReference)
+  }
 
   @PostMapping(Array("/updateRunStatus/{uniqueId}"))
-  def updateRunStatus(): CompletableFuture[RunStatus] = Future(null)
+  @ResponseStatus(HttpStatus.OK)
+  def updateRunStatus(@PathVariable uniqueId: String,
+                      @RequestBody runStatus: RunStatus): CompletableFuture[Run] = {
```

Comment:
shouldn't be like this: 
`{
        "status": {
            "enumClass": "za.co.absa.atum.model.RunState",
            "value": "stageSucceeded"
        }
} `